### PR TITLE
Rename rfactor domain as logical domain

### PR DESCRIPTION
**Pull request by zasdfgbnm**

### CODE REVIEW

File: csrc/ir/internal_base_nodes.h [hunk 1]
```diff
@@ -553,7 +553,7 @@ class TensorDomain : public Val {
     return !allocation_domain_.empty();
   }
 
-  // Returns if rfactor domain only consists of id's of iter type.
+  // Returns if logical domain only consists of id's of iter type.
   bool hasViewLikeRFactor() const;
```
Comment (zasdfgbnm) [inline csrc/ir/internal_base_nodes.h:557]: I am not renaming this in this PR. I do not have a good name for it, especially, I don't think `hasViewLikeLogical` is not a good name. I will leave it like this. Likely, in the future, I will rename it as `hasViewLikeProducerProjection` together with renaming the root domain as producer projection.

**Comment by zasdfgbnm** [comment]
!build --diff --diff-bench

**Comment by zasdfgbnm** [comment]
!build

**Review by naoyam — APPROVED** [review]
Thanks for the cleanup! LGTM.

**Pull request merged**